spring-batch-integration/src/test/java/org/springframework/batch/integration/partition/RemotePartitioningMasterStepBuilderTest.java

```diff
@@ -0,0 +1,273 @@
+		this.expectedException.expectMessage("outputChannel must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.outputChannel(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void messagingTemplateMustNotBeNull() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("messagingTemplate must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.messagingTemplate(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void jobExplorerMustNotBeNull() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("jobExplorer must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.jobExplorer(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void pollIntervalMustBeGreaterThanZero() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("The poll interval must be greater than zero");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.pollInterval(-1)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void timeoutMustNotBeGreaterThanZero() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("The timeout must be greater than zero");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.timeout(-1L)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void testMandatoryOutputChannel() {
+		// given
+		RemotePartitioningMasterStepBuilder builder = new RemotePartitioningMasterStepBuilder("step");
+
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("An outputChannel must be provided");
+
+		// when
+		Step step = builder.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void testUnsupportedOperationExceptionWhenSpecifyingPartitionHandler() {
+		// given
+		PartitionHandler partitionHandler = Mockito.mock(PartitionHandler.class);
+		this.expectedException.expect(UnsupportedOperationException.class);
+		this.expectedException.expectMessage("When configuring a master step " +
+				"for remote partitioning using the RemotePartitioningMasterStepBuilder, " +
+				"the partition handler will be automatically set to an instance " +
+				"of MessageChannelPartitionHandler. The partition handler must " +
+				"not be provided in this case.");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.partitionHandler(partitionHandler)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void testMasterStepCreation_whenPollingRepository() {
```

Comment:
Why camel case and _?  I'd rather just keep the camel case only.

Reply:
fixed.